src/main/java/org/terasology/crashreporter/pages/UserInfoPanel.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.crashreporter.pages;
+
+import org.terasology.crashreporter.I18N;
+import org.terasology.crashreporter.Resources;
+
+import javax.swing.*;
+import javax.swing.border.EmptyBorder;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import java.awt.*;
+import java.nio.file.Path;
+
+/**
+ * The panel that requests some additional info from the user
+ */
+public class UserInfoPanel extends JPanel {
+    private static final long serialVersionUID = -1714121731354180405L;
+
+    private JLabel requestLabel;
+    private final JTextArea userMessageArea;
+    private JLabel statusLabel;
+
+    private String log;
+    private final Path logFile;
+
+    private Boolean additionalInfoPresent;
+
+    /**
+     * @param log     the current contents of the log file
+     * @param logFile the location of the relevant file
+     */
+    public UserInfoPanel(String log, Path logFile) {
+        this.log = log;
+        this.logFile = logFile;
+        additionalInfoPresent = false;
+
+        setLayout(new BorderLayout(50, 20));
+
+        String htmlText = "<html><h3>" + I18N.getMessage("additionalInfoTitle") + "</h3>" + I18N.getMessage("additionalInfoMessage") + "</html>";
+
+        requestLabel = new JLabel(htmlText, Resources.loadIcon("icons/Actions-dialog-info-icon.png"), SwingConstants.CENTER);
+        requestLabel.setBorder(new EmptyBorder(10, 0, 0, 0));
+
+        userMessageArea = new JTextArea();
+        userMessageArea.setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
+        userMessageArea.setFont(getFont().deriveFont(Font.PLAIN).deriveFont(12f));
+
+        userMessageArea.getDocument().addDocumentListener(new DocumentListener() {
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                additionalInfoPresent = true;
+                changeStatus();
```

Comment:
I'd rather rename it to `refreshStatus` and check whether the document is empty or not in that method. The `additionalInfoPresent` flag is obsolete then.